src/pages/SetPassword.jsx

```diff
@@ -0,0 +1,113 @@
+import EmailIcon from "@mui/icons-material/Email";
+import { useRequest } from "ahooks";
+import { useForm } from "react-hook-form";
+import { toast } from "react-hot-toast";
+import { Navigate, useNavigate } from "react-router-dom";
+
+import { useAuth } from "@/models";
+
+export default function SetPassword() {
+  const auth = useAuth();
+  const { register, handleSubmit } = useForm();
+  const navigate = useNavigate();
+
+  if (auth.isSignedIn()) {
+    return <Navigate to="/" replace />;
+  }
+
+  // follow this function for a better practice with interactive button
+  const { run: sendPasswordResetEmail, loading: Loading } = useRequest(
+    async (data) => {
+      await auth.resetPassWordByEmail(data.email);
+      return data.email;
+    },
+    {
+      manual: true,
+      onSuccess: async (email) => {
+        toast.success("Email Sent!");
+        const emailToPass = { email: email };
+        navigate("/confirm_send_email", { state: emailToPass });
+      },
+      onError: (error) => {
+        if (error.code === "auth/user-not-found") {
+          toast.error("Failed: Email hasn't been regsitered by RefugeeOne!");
+        } else if (error.code === "auth/invalid-email") {
+          toast.error("Failed: This is not a valid email!");
+        } else {
+          // firebase has limits on numbers to reset password. Hard to include all error codes
+          toast.error("Sorry, something is wrong. Please come back later.");
+        }
+        console.log(error.code);
+      },
+    }
+  );
+
+  return (
+    <div className="flex flex-col justify-center items-center">
+      <div className="card max-w-md">
+        <div className="card-body">
+          <p className="mb-5 text-3xl">Welcome!</p>
+          <p className="mb-1 text-xl">Please enter your email</p>
+          <p className="mb-1 text-sm">
+            * Please make sure your email has been registered by the RefugeeOne team. Once
+            you enter a registed email, we will send you a link to set your account
```

Comment:
Can we add a info icon before all the notes

Reply:
fixed